src/refs.c

```diff
@@ -927,6 +926,12 @@ static int ensure_segment_validity(const char *name)
 		if (prev == '@' && *current == '{')
 			return -1; /* Refname contains "@{" */
 
+		if (*current == '*') {
+			if (!may_contain_glob)
+				return -1;
```

Comment:
I realize that you didn't introduce this problem, but reviewing your change shows that we might have one with error reporting.  We return `-1` out of this function w/o setting an error message.  When called by `git_reference__normalize_name`, it will set an error message iff the error code is `GIT_EINVALIDSPEC`.  So it looks to me like we'll return an error without setting a message.  I think we want to return `GIT_EINVALIDSPEC` from this function on any error?

Reply:
Nope, in fact we're good, even though it's non-obvious. First note that we do not set any error code via `ensure_segment_validity` at all, but only the segment length. But we do initialize `error = GIT_EINVALIDSPEC` in the calling function and make sure that it doesn't get set to a different value throughout the function.

Reply:
Cool.  Thanks for the explanation.